From b44857d7d97fd83fc4c0f58d3a660d1988ed3d11 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:06 -0400
Subject: [tools] perf: Introduce filename__read_int helper

Message-id: <1408462094-14194-198-git-send-email-jolsa@redhat.com>
Patchwork-id: 88208
O-Subject: [PATCH RHEL7.1 BZ1131394 197/325] perf tools: Introduce filename__read_int helper
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 97a07f10c38064dea492794c99445f6260afcdc1
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Thu Oct 17 16:33:43 2013 -0300

description
===========
Just opens a file and calls atoi() in at most its first 64 bytes.

To read things like /proc/sys/kernel/perf_event_paranoid.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/util.c b/tools/perf/util/util.c
index 8dc8cf3..c25e57b 100644
--- a/tools/perf/util/util.c
+++ b/tools/perf/util/util.c
@@ -394,3 +394,20 @@ unsigned long parse_tag_value(const char *str, struct parse_tag *tags)
 
  return (unsigned long) -1;
 }
+
+int filename__read_int(const char *filename, int *value)
+{
+ char line[64];
+ int fd = open(filename, O_RDONLY), err = -1;
+
+ if (fd < 0)
+  return -1;
+
+ if (read(fd, line, sizeof(line)) > 0) {
+  *value = atoi(line);
+  err = 0;
+ }
+
+ close(fd);
+ return err;
+}
diff --git a/tools/perf/util/util.h b/tools/perf/util/util.h
index 42dfba7..c8f362d 100644
--- a/tools/perf/util/util.h
+++ b/tools/perf/util/util.h
@@ -305,4 +305,6 @@ struct dso;
 
 char *get_srcline(struct dso *dso, unsigned long addr);
 void free_srcline(char *srcline);
+
+int filename__read_int(const char *filename, int *value);
 #endif /* GIT_COMPAT_UTIL_H */
-- 
1.7.1